frontend/src/metabase/actions/containers/ActionParametersInputForm/ActionParametersInputForm.unit.spec.tsx

```diff
@@ -0,0 +1,228 @@
+import React from "react";
+import _ from "underscore";
+import nock from "nock";
+import userEvent from "@testing-library/user-event";
+import { waitFor } from "@testing-library/react";
+
+import { render, screen } from "__support__/ui";
+
+import {
+  createMockActionParameter,
+  createMockQueryAction,
+  createMockImplicitQueryAction,
+} from "metabase-types/api/mocks";
+
+import ActionParametersInputForm, {
+  ActionParametersInputModal,
+} from "./ActionParametersInputForm";
+
+const defaultProps = {
+  missingParameters: [
+    createMockActionParameter({
+      id: "1",
+      name: "Parameter 1",
+      type: "type/Text",
+    }),
+    createMockActionParameter({
+      id: "2",
+      name: "Parameter 2",
+      type: "type/Text",
+    }),
+  ],
+  dashcardParamValues: {},
+  action: createMockQueryAction(),
+  dashboard: { id: 123 },
+  dashcard: { id: 456 },
+  onSubmit: jest.fn(() => ({ success: true })),
+  onCancel: _.noop,
+  onSubmitSuccess: _.noop,
+};
+
+async function setup(options?: any) {
+  render(<ActionParametersInputForm {...defaultProps} {...options} />);
+}
+
+async function setupModal(options?: any) {
+  render(
+    <ActionParametersInputModal
+      title="Test Modal"
+      onClose={_.noop}
+      {...defaultProps}
+      {...options}
+    />,
+  );
+}
+
+function setupPrefetch() {
+  nock(location.origin)
+    .get(uri => uri.includes("/api/dashboard/123/dashcard/456/execute"))
+    .reply(200, {
+      "1": "uno",
+      "2": "dos",
+    });
+}
+
+describe("Actions > ActionParametersInputForm", () => {
```

Comment:
This test suite throws a `You seem to have overlapping act() calls, this is not supported. Be sure to await previous act() calls before making a new one` error. Could you please investigate? It might be related to awaited `userEvent` calls or maybe async functions inside `waitFor`

Reply:
looks like it was a consequence of using `await findByX()` within `await watiFor()`